### DISCUSSION TITLE
Error when using ST::transform(): could not parse proj string '4326'

### DISCUSSION THREAD

**Discussion by BezBIS** [clickbar/laravel-magellan#90] [Q&A] [answered]
I am getting an error when trying to pass a srid to `ST::transform()` as part of a select statement. A minimal example is:
```
$geom = Site::stSelect(ST::transform('geom', 4326))->first();
```

and the error is 
```
SQLSTATE[XX000]: Internal error: 7 ERROR: could not parse proj string '4326' CONTEXT: SQL function "st_transform" statement 1
```

Running the generated query through pgAdmin works correctly, as does using `selectRaw`. I want to avoid selectRaw, however, as I want to get the extent of the transformed geometry as a Box2D:
```
$bounds = Site::stSelect(ST::extent(
    ST::transform('geom', 4326)
))
    ->where('id', '=', $search->params['site'])
    ->first();
```

As I'm fairly new to the library, it's entirely possible I'm missing something. That being the case, I thought I would ask fr help as a discussion before opening a bug report.

**Comment by saibotk** [comment]
Oh, interesting!

Thanks for opening up the discussion, it seems like it tries to use another overload, where it expects the projection string.

Can you show me the generated SQL?

Otherwise, I would need some more time until i can try this out in my test setup.

**Reply by BezBIS** [reply]
Thank you for the very quick response.
The generated SQL for the minimal query is: `select ST_Transform("geom"::geometry, 4326) AS transform from "sites" limit 1`. Copy and pasting this into pgAdmin for the same database works fine, so I'm slightly confused as to where the error is originating from.

**Reply by saibotk** [reply]
Mhhh, is this the sql output from the stSelect example using `->toSql()`?

Seems very odd, my initial expectation was that it just mistakenly cast the number to string, but that does not seem to be the case.

**Reply by BezBIS** [reply]
Aha, sorry I was copying the generated SQL from debugbar. The actual output of `toSql()` is: `select ST_Transform("geom"::geometry, ?) AS transform from "sites"`.

**Reply by saibotk** [reply, marked as answer]
Okay weird, I reproduced this. I don't really see any obvious mistake on our side here, since the raw sql looks perfectly fine.

Seems like we gotta dive a bit deeper under the hood of Laravel for this :( 

In any case this is a valid bug, so i will create an issue from this.
We try to get this resolved as soon as possible, but i cannot make any guarantees because im currently busy with uni stuff, might look into it when i get some sparetime in the evenings! Just as a heads up!

**Reply by saibotk** [reply]
See #91

**Reply by BezBIS** [reply]
No worries at all and thank you for looking into it. This is required for a very minor feature in an R&D project I'm working on at the moment, so not at all critical. Thank you for your work on the library, I've not had a proper chance to look into it yet, but it seems very useful.
